algebra-core/src/msm/variable_base.rs

```diff
@@ -39,7 +41,8 @@ impl VariableBaseMSM {
             .map(|w_start| {
                 let mut res = zero;
                 // We don't need the "zero" bucket, so we only have 2^c - 1 buckets
-                let mut buckets = vec![zero; (1 << c) - 1];
+                let log2_n_bucket = if (w_start % c) != 0 { w_start % c } else { c };
```

Comment:
Interesting, why this change?

Reply:
it probably doesn't make a big difference. But still, let's say that you had just a few bits for the last window decomposition for the MSM. It could help a little with the memory allocation. 

Its just slightly more correct, anyway, if a nitpick

Reply:
Oh, I see, that makes sense.